### PR TITLE
fix: correctly select properties of primitives

### LINKED ISSUE #1245: Validation on nested fields not working.
## Describe the bug

For some inputs, validation on nested fields accessed by dot (.) is not returning error
This was working in v6 but changed in  v7

## To Reproduce

Please check https://runkit.com/tony-sunny/6503695da7894b0008762533

## Expected behavior

Expected validation chain to return error

## Current behavior

There are no errors in validaiton result

## Environment:

- Express-validator version:  7.0.1
- Express version: 4.18.2
- Node.js version: 18.16.1

### PR DESCRIPTION
**Pull request by fedeci**
## Description

Fixes #1245 
It is correct to assume that primitives will not have properties in a request.
Despite that express-validator should provide the correct result, so if a user wants to validate a non existent property we should treat that as undefined.
Eventually in the future we may also update `FieldInstance` to look something like this:

```typescript
export interface FieldInstance {
  path: string;
  originalPath: string;
  location: Location;
  value: any;
  exists: boolean // added property
}
```

## To-do list

- [x] I have added tests for what I changed.

- [x] This pull request is ready to merge.

### CODE REVIEW

File: src/express-validator.spec.ts
Comment (fedeci) [inline src/express-validator.spec.ts:1]: Do we have another place to store these tests?

File: src/field-selection.ts [hunk 1]
```diff
@@ -52,13 +52,19 @@ function expandPath(object: any, path: string | string[], currPath: readonly str
...
+      return [];
     }
-
-    // there still are paths to traverse, but value is a primitive, stop
-    return [];
+    if (key === '*') {
+      // wildcard position does not exist
+      return [];
+    }
+    // value is a primitive, there are still still paths to traverse that will be likely undefined, we return the entire path
```
Comment (gustavohenke) [inline src/field-selection.ts:66]: Fun fact: you could theoretically go forever now (not for any good reason though), e.g. `foo.toFixed.name.toString.length.toFixed`... you just have to keep diving deeper into the prototype!

```suggestion
    // value is a primitive, paths being traversed from here might be in their prototype, return the entire path
```

File: src/field-selection.spec.ts [hunk 2]
```diff
@@ -113,14 +115,19 @@ describe('selectFields()', () => {
     const req = {
       query: { foo: ['bar', 'baz'] },
     };
-    const instances = selectFields(req, ['foo[1]'], ['query']);
+    const instances = selectFields(req, ['foo[1]', 'foo[2]'], ['query']);
```
Comment (gustavohenke) [inline src/field-selection.spec.ts:118]: OOC, does this cover anything non-obvious, or is just for thoroughness?

File: src/express-validator.spec.ts
Comment (gustavohenke) [reply src/express-validator.spec.ts:1]: The tested aspects are the field selection + `exists` functionality.
I don't feel like they are needed though, everything is covered in some way already.

**Pull request merged**